src/software/simulation/physics_simulator_ball.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+
+#include "software/simulation/physics/physics_ball.h"
+#include "software/simulation/simulator_ball.h"
+
+/**
+ * The SimulatorBall class acts as a wrapper for a PhysicsBall that deals with more
+ * logic-focused elements for simulation.
+ */
+class PhysicsSimulatorBall : public SimulatorBall
+{
+   public:
+    /**
+     * Create a new PhysicsSimulatorBall given a PhysicsBall
+     *
+     * @param physics_ball the PhysicsBall to simulate and control
+     */
+    explicit PhysicsSimulatorBall(std::weak_ptr<PhysicsBall> physics_ball);
+    explicit PhysicsSimulatorBall() = delete;
+
+    /**
+     * Returns the current position of the ball, in global field coordinates, in meters
+     *
+     * @return the current position of the ball, in global field coordinates, in meters
+     */
+    Point position() const override;
+
+    /**
+     * Returns the current velocity of the ball, in global field coordinates, in m/s
+     *
+     * @return the current velocity of the ball, in global field coordinates, in m/s
+     */
+    Vector velocity() const override;
```

Comment:
nit: javadoc comments aren't necessary for overridden functions

Reply:
Removed